include/zephyr/bluetooth/gatt.h

```diff
@@ -626,7 +626,7 @@ ssize_t bt_gatt_attr_read_service(struct bt_conn *conn,
  */
 #define BT_GATT_PRIMARY_SERVICE(_service)				\
 	BT_GATT_ATTRIBUTE(BT_UUID_GATT_PRIMARY, BT_GATT_PERM_READ,	\
-			 bt_gatt_attr_read_service, NULL, _service)
+			 bt_gatt_attr_read_service, NULL, (void *)_service)
```

Comment:
the field user_data type `void *`, as this PR change all Service UUID to rodata, so there must add `(void *)` to avoid warning.

Reply:
I take this opportunity to explain why I want to do this PR. Since Service UUID and Char UUID should not change in the service definition, they are most reasonably defined as rodata, also for save some ram footprint.

Reply:
Ok, makes sense. Can you add this info to the commit message?

Reply:
Only `subsys/bluetooth/host/gatt.c:1761` will use this user_data, so it's safe.